hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -239,7 +250,7 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
 
     try {
       Table table = clients.run(client -> client.getTable(fromDatabase, fromName));
-      HiveOperationsBase.validateTableIsIceberg(table, fullTableName(name, from));
+      validateTableIsIcebergTableOrView(contentType, name, table, from);
```

Comment:
rather than passing the catalog name / table / from separately, it's probably better to just pass the table and the fullTableName: `fullTableName(name, from)`